frontend/src/app/components/add-post-modal/add-post.component.html

```diff
@@ -4,11 +4,11 @@ <h1 mat-dialog-title>Add Post</h1>
     <mat-label>Title</mat-label>
     <input
       name="title"
+      [appAutofocus]="true"
       [formControl]="titleControl"
       [errorStateMatcher]="matcher"
       matInput
       autocomplete="off"
-      cdkFocusInitial
```

Comment:
So why are we not just using `cdkFocusInitial` for the initial autofocus? What does `[appAutofocus]` do that's better?

Reply:
With the cdkFocusInitial it pings the mat-error and makes the title field red when you open the new post modal. I tried to add more conditions like making sure the mat-error triggers once the user has actually typed something, but it still gave the same error, so i went with the directive.
The directive is very simple and straightforward and basically does what the "stock" autofocus html option does (but the problem with that is it only works the first time you open a modal, and not for the times after you have opened the modal once).

Reply:
If the issue persists after the first time you open the modal, then it's probably not the best solution.

Can you maybe try what's suggested here: https://stackoverflow.com/questions/51503940/how-to-avoid-mat-form-field-validation-trigger-when-matinput-lose-focus

Reply:
So you want to drop the directive?

Reply:
Yeah that would be nice since ideally we want a solution that works when the modals opened multiple times.

Reply:
no the directive works perfectly, i mentioned 'autofocus' since its a built in option in html that only works the first time. That lead me to make the `[autoFocus]` directive which works like we want it too.

Reply:
j re-read my comment and it sounds like i was talking about the directive only working once xd. i meant the html autofocus.

Reply:
Oh gotcha, yeah I misinterpreted your comment. I'm good to keep this then.